controllers/auth.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const { validate } = require('indicative').validator;
+const { AuthorizationError } = require('errors');
+
+const FreedomHelper = require('helpers/freedom_helper');
+const User = require('models/User');
+
+const jwt = require('lib/jwt');
+const responses = require('lib/responses');
+
+module.exports = {
+    login,
+    logout
+};
+
+async function login (req) {
+    const data = req.body;
+
+    const rules = {
+        code: 'required|string',
+        redirect_uri: 'required|url',
+        grant_type: 'required|string'
+    };
+
+    const validated_data = await validate(data, rules);
+    let oauth_response = {};
+
+    try {
+        oauth_response = await FreedomHelper.get_oauth_access_token(validated_data);
+    }
+    catch (err) {
```

Comment:
I think we don't need try-catch here. The router_extended.js will already handle this.

```
await validate(..);

let oauth_response = await FreedomHelper.get(...)

if (!oauth_response.access_token) {
   throw new ...
}
```

Reply:
Removed the `try/catch`